kadena/light-client/src/types/chainweb.rs

```diff
@@ -55,3 +61,54 @@ impl TryInto<Spv> for SpvResponse {
         Ok(Spv::new(self.algorithm, self.chain, object, subject))
     }
 }
+
+/// Response received while querying a  Payload from a Chainweb
+/// node.
+#[derive(Clone, Debug, Deserialize, Getters)]
+#[serde(rename_all = "camelCase")]
+#[getset(get = "pub")]
+pub struct PayloadResponse {
+    coinbase: String,
+    miner_data: String,
+    outputs_hash: String,
+    payload_hash: String,
+    transactions: Vec<Vec<String>>,
+    transactions_hash: String,
+}
+
+impl PayloadResponse {
+    /// Get the transaction output key for the given transaction index.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_index` - The index of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// The transaction output key.
+    pub fn get_transaction_output_key(&self, transaction_index: usize) -> Result<String> {
```

Comment:
This name makes it sound like you're getting some output of the transaction, but the request key is more like the transaction hash or ID. You can get the request key also from the transaction input, since it is also available in the transaction json (if you `.get(0)` below instead of `.get(1)`) as the `"hash"` property.

I'm ok with keeping this name if you want, but just mentioning that I don't think "output key" is a term anyone else is using to refer to this piece of data

Reply:
I'll rename it `get_transaction_request_key` as it seems to be the naming convention